### ISSUE TITLE
All Python code exits with a `SystemError` after an exception has occurred

### ISSUE THREAD

**Issue by Samyak2** [metacall/core#261]
## 🐛 Bug Report

When running python code through the CLI, once there's a python error (for example, a `TypeError`), all subsequent python calls, even to the same function, fail with:
```
Python Error: [Type: <class 'SystemError'>]: _PyEval_EvalFrameDefault returned a result with an error set
```

### Expected Behavior

Subsequent correct python code should run as expected.

### Current Behavior

No python code runs after an exception has occurred,

### Possible Solution

The error/exception needs to be cleared. Not sure how.

### Steps to Reproduce

1. Create a python file named `hello.py` with:
    ```python
    def add(a, b):
        return a + b

    print("Hello world")
    ```
2. Run the following in `metacallcli`:
    ```
    load py hello.py
    call add(5, 10)
    call add(1, "a")
    call add(1, 1)
    ```
4. Notice that the first `call` runs correctly giving the output `15`
5. The 2nd `call` fails with: 
    ```
    Error: Python Error: [Type: <class 'TypeError'>]: unsupported operand type(s) for +: 'int' and 'str'
    Traceback not available
    ```
7. The 3rd and all subsequent calls fail with:
     ```
     Error: Python Error: [Type: <class 'SystemError'>]: _PyEval_EvalFrameDefault returned a result with an error set
    Traceback not available
    (null)
     ```

### Context (Environment)

I built metacall from source with python loader support. I was trying out metacall with a simple python example when this issue occurred.

### Detailed Description

N/A

### Possible Implementation

N/A

**Comment by viferga** [comment]
We can add a small patch in order to solve this, just clearing the error after the call.

But I think the best way of solving this is solving this issue: https://github.com/metacall/core/issues/141

Catching the exception, propagating it to the top module and clearing it from the runtime.

**Comment by Samyak2** [comment]
> We can add a small patch in order to solve this, just clearing the error after the call.
> 
> But I think the best way of solving this is solving this issue: #141
> 
> Catching the exception, propagating it to the top module and clearing it from the runtime.

I don't have any experience with this codebase, so can I try making a patch to fix this for now? I can help with #141 once I'm more familiar with the codebase and the context around the issue.

**Comment by viferga** [comment]
I have solved the issue, this is how it looks like now:
```
96: λ {"ExceptionThrown":{"message":"unsupported operand type(s) for +: 'int' and 'str'","label":"TypeError","code":0,"stacktrace":"Traceback not available"}}
96: λ 66
96: λ Exiting ...
```

**Comment by viferga** [comment]
> > We can add a small patch in order to solve this, just clearing the error after the call.
> > But I think the best way of solving this is solving this issue: #141
> > Catching the exception, propagating it to the top module and clearing it from the runtime.
> 
> I don't have any experience with this codebase, so can I try making a patch to fix this for now? I can help with #141 once I'm more familiar with the codebase and the context around the issue.

Respect to this, I just have added support for exception handling but there is a list of TODO. I can make a diff so you can check what I have done and implement the same for other languages. I am closing this issue for now, and I am going to add the TODO list in https://github.com/metacall/core/issues/141
